pkg/operator/ceph/object/bucket/rgw-handlers.go

```diff
@@ -37,15 +78,18 @@ func (p *Provisioner) createCephUser(username string) (accKey string, secKey str
 		if errors.Is(err, admin.ErrNoSuchUser) {
 			u, err = p.adminOpsClient.CreateUser(p.clusterInfo.Context, userConfig)
 			if err != nil {
-				return "", "", errors.Wrapf(err, "failed to create ceph object user %v", userConfig.ID)
+				return "", "", false, errors.Wrapf(err, "failed to create Ceph object user %v", userConfig.ID)
 			}
+			newUser = true
 		} else {
-			return "", "", errors.Wrapf(err, "failed to get ceph user %q", username)
+			return "", "", false, errors.Wrapf(err, "failed to get Ceph object user %q", username)
 		}
+	} else {
+		logger.Infof("Ceph object user %q already exists", username)
 	}
 
-	logger.Infof("successfully created Ceph user %q with access keys", username)
-	return u.Keys[0].AccessKey, u.Keys[0].SecretKey, nil
+	logger.Infof("successfully created Ceph object user %q with access keys", username)
+	return u.Keys[0].AccessKey, u.Keys[0].SecretKey, newUser, nil
```

Comment:
This logic only works if the user-create-then-process is atomic, which it isn't. If the new user is created and then the reconcil fails, the next reconcile will have an incorrect understanding of whether the user is truly new. I think we need a different heuristic to control the behavior that gets changed by this `newUser` result.

Reply:
I wrestled with how to handle this. The only other method I came up with was to annotate the rgw user in some way such as setting a value into the display_name or email field that would indicate that it was an operator created user.  Do you want me to switch to that approach?

Reply:
What do you think about only modifying users which match `genUserName(obc)` as I did in `deleteOBUser`?

Reply:
As we don't actually care if it is a new user... that was just a proxy to try to determine if it was an externally managed user (or not).

Reply:
I've removed the usage of `newUser` completed and `genUserName()` / if the user name looks like an operator generated user name is now the only heuristic used.